src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.SpeedOpt.cs

```diff
@@ -15,7 +15,7 @@ private abstract partial class OrderedIterator<TElement>
             public override TElement[] ToArray()
             {
                 TElement[] buffer = _source.ToArray();
-                if (buffer.Length == 0)
+                if (buffer.Length <= 1)
                 {
                     return buffer;
```

Comment:
Won't this be problematic if the user decides to mutate the array contents? I'd expect ToArray to return a new array then.

Reply:
Why would it be problematic if the consumer mutates the returned array?

Reply:
Oh wait, `_source.ToArray()` copies the array, missed the `ToArray` part and thought it might return the array passed to the API.